### PR TITLE
List new releases

### PR DESCRIPTION
**Pull request by emosbaugh**

### CODE REVIEW

File: pkg/upstream/peek.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	}
+	if u.Scheme == "helm" {
+		return peekHelm(u, fetchOptions.HelmRepoURI)
+	}
+	if u.Scheme == "replicated" {
+		return peekReplicated(u, fetchOptions.LocalPath, fetchOptions.License, fetchOptions.CurrentCursor)
+	}
+	if u.Scheme == "git" {
+		// return peekGit(upstreamURI)
+		// TODO
```
Comment (emosbaugh) [inline pkg/upstream/peek.go]: TODO

File: pkg/upstream/peek.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	if !util.IsURL(upstreamURI) {
+		return nil, errors.New("not implemented")
+	}
+
+	u, err := url.ParseRequestURI(upstreamURI)
+	if err != nil {
+		return nil, errors.Wrap(err, "parse request uri failed")
+	}
+	if u.Scheme == "helm" {
+		return peekHelm(u, fetchOptions.HelmRepoURI)
```
Comment (emosbaugh) [inline pkg/upstream/peek.go]: TODO

File: pkg/upstream/peek.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	if u.Scheme == "replicated" {
+		return peekReplicated(u, fetchOptions.LocalPath, fetchOptions.License, fetchOptions.CurrentCursor)
+	}
+	if u.Scheme == "git" {
+		// return peekGit(upstreamURI)
+		// TODO
+	}
+	if u.Scheme == "http" || u.Scheme == "https" {
+		// return peekHttp(upstreamURI)
+		// TODO
```
Comment (emosbaugh) [inline pkg/upstream/peek.go]: TODO

File: pkg/pull/peek.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	Namespace     string
+	LocalPath     string
+	LicenseFile   string
+	CurrentCursor string
+	Silent        bool
+}
+
+// Peek will retrieve all later versions of the application specified in upstreamURI
+// using the options specified in peekOptions. It returns a list of versions.
+func Peek(upstreamURI string, peekOptions PeekOptions) ([]upstream.Update, error) {
```
Comment (divolgin) [inline pkg/pull/peek.go]: Why is this `Peek` and not `GetUpdates`?  This needs a better name or a revised comment about what it does.

**Pull request merged**